xmodule/x_module.py

```diff
@@ -1705,6 +1677,19 @@ def rebind_noauth_module_to_user(self):
         if rebind_user_service:
             return partial(rebind_user_service.rebind_noauth_module_to_user)
 
+    @property
+    def course_id(self):
+        """
+        Old API to get the course ID.
+
+        Deprecated in favor of `runtime.scope_ids.usage_id.context_key`.
+        """
+        warnings.warn(
+            "`runtime.course_id` is deprecated. Use `context_key` instead: `runtime.scope_ids.usage_id.context_key`.",
+            DeprecationWarning, stacklevel=3,
+        )
+        return self.descriptor_runtime.course_id.for_branch(None)
```

Comment:
@jristau1984: Sometime after this PR merges, you might want to look for this line in the Splunk logs. In particular, this could happen if edx.org uses extensions or XBlocks that try to access this deprecated attribute directly instead of deriving it directly from the UsageKey.